molstar/example-gallery
# Switch from CodePen to CodeSandbox w/ 2 examples

## Overview
This PR proposes a change from CodePen to CodeSandbox for the better developer experience when trying out the examples. CodeSandbox has a much better editor and builds the project live, allowing examples to be written in TS and loaded directly from the GitHub repository.

## Changes made
- Changed README to describe new process for testing the examples on codesandbox and locally
- Created new bare-bones schema for examples in `./default`
    - `./default` can also be used as a blank template for trying out mol*
- Created selection example "select_ligand_and_surroundings"
- Created representation example "transparency_using_selection"

## How to test
1. Using codesandbox, navigate to the 3 following URLs using an incognito browser. Ensure the sandbox is functional and the behavior is properly displayed in the viewer.
    - [Default](https://codesandbox.io/p/sandbox/github/molstar/example-gallery/tree/codesandbox/default)
    - [Ligand with surrounding](https://codesandbox.io/p/sandbox/github/molstar/example-gallery/tree/codesandbox/selection/select_ligand_and_surroundings)
    - [Set transparency on selection](https://codesandbox.io/p/sandbox/github/molstar/example-gallery/tree/codesandbox/representation/transparency_using_selection)

2. Navigate to one of the previous URLs in a regular browser and sign in to codesandbox (using any account). Ensure you can edit the code and the changes are reflected in the viewer.
3. Locally, `cd ./default ` to one of the example folders and build locally:
```sh
npm install
npm run start
```

## Specific review requests
- Is the process of opening a codesandbox example and testing it out straightforward and easy?
- For each of the examples:
  - Are there enough comments to sufficiently describe the process?
  - Are the comments accurately describing the process?
  - Are there any anti-patterns in the examples?
  - Can the examples be more efficient?
